pkg/api/meta.go

```diff
@@ -25,6 +25,9 @@ import (
 // FillObjectMetaSystemFields populates fields that are managed by the system on ObjectMeta.
 func FillObjectMetaSystemFields(ctx Context, meta *ObjectMeta) {
 	meta.CreationTimestamp = util.Now()
+	if user, ok := UserFrom(ctx); ok {
```

Comment:
There could not be a user if it's an object created before this change went
in, created by a core system process (like endpoints and namespaces created
by the master on startup), or created by direct edit in etcd that does not
require authentication.  Or a cluster that doesn't require authentication
or doesn't pass that info to Kube.

On Mon, Sep 14, 2015 at 2:08 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> Yes, I don't want to create new names even if the english is slightly
> weird.  We're fairly good at "<concept><type>" in our API - "creation" is
> the concept, "type" is "UserName", "Timestamp", "PeriodInSeconds", "Name",
> "Ref"
>
> On Mon, Sep 14, 2015 at 1:41 PM, Derek Carr <notifications@github.com>
> wrote:
>
>> In pkg/api/meta.go
>> <https://github.com/kubernetes/kubernetes/pull/13925#discussion_r39422744>
>> :
>>
>> > @@ -25,6 +25,9 @@ import (
>> >  // FillObjectMetaSystemFields populates fields that are managed by the system on ObjectMeta.
>> >  func FillObjectMetaSystemFields(ctx Context, meta *ObjectMeta) {
>> >  	meta.CreationTimestamp = util.Now()
>> > +	if user, ok := UserFrom(ctx); ok {
>>
>> When would there never be a user?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/13925/files#r39422744>.
>>
>
>
>
> --
> Clayton Coleman | Lead Engineer, OpenShift
>



-- 
Clayton Coleman | Lead Engineer, OpenShift